src/common/TrackedOp.h

```diff
@@ -190,19 +190,35 @@ class OpTracker {
   {
     typename T::Ref retval(new T(params, this));
     retval->tracking_start();
-
     if (is_tracking()) {
-      retval->mark_event("header_read", params->get_recv_stamp());
-      retval->mark_event("throttled", params->get_throttle_stamp());
-      retval->mark_event("all_read", params->get_recv_complete_stamp());
-      retval->mark_event("dispatched", params->get_dispatch_stamp());
+      struct MarkEventStore {
+	std::string str;
+	utime_t stamp;
+
+	MarkEventStore(std::string s, utime_t t) : str(s), stamp(t) {}
+      };
+
+      auto compare = [] (auto first, auto second) -> bool {
```

Comment:
i am confused. why do we need to sort by the timestamp? as i think the events are always triggered in chronological order. like

1. header_read
2. throttled
3. all_read
4. dispatched

Reply:
Agreed.  Also, we really don't want to sort here in any event for the same reasons as above.

Reply:
In the dump_historic_ops, we got the resulting stamp for throttled is earlier to header_read; and then on later test, also to initiated timestamp; Josh suggested as they are coming straight from messenger, this can happen.
`    [
                        "initiated",
                        "2019-06-28T16:22:45.312798-0400",
                        0
                    ],
                    [
                        "header_read",
                        "2019-06-28T16:22:45.312798-0400",
                        4294967295.9938254
                    ],
                    [
                        "throttled",
                        "2019-06-28T16:22:45.306623-0400",
                        0.0062214990000000001
                    ],
                    [
                        "all_read",
                        "2019-06-28T16:22:45.312844-0400",
                        1.1671000000000001e-05
                    ],
  `

Reply:
I always got throttled stamp earlier than header_read, will updating their order alone would be better as per keeping optimization in mind?